### PR TITLE
Cypress e2e tests for the About Us page

### PR DESCRIPTION
**Pull request by TimeaKovacs86**
### What does this pull request aim to achieve? 💯
- Cover the About Us page with end-to-end tests

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

### CODE REVIEW

**Comment by TimeaKovacs86** [comment]
Hi everyone,
this is my first test ever, could you please give me some feedback on it? Appreciate your help!

**Comment by adorapham** [comment]
Hi @TimeaKovacs86 , **1**-so far I see you have the function call to get element out of the class.  the class AboutUs has only data, it seem like a JSON data file to me. **2**- I am not sure if we should check the the links of supporters (life-time, current and past), might be we need to confirm with @galeza and @jessica-lau   **3**- I don't see the check on email links. TY

File: cypresse2e/cypress/elements/pages/aboutUs.js
Comment (jessica-lau) [inline cypresse2e/cypress/elements/pages/aboutUs.js]: 'E' in 'goToHElpBoardLink' should be lowercased

File: cypresse2e/cypress/integration/e2eTests/aboutUs/aboutUs.spec.js
Comment (jessica-lau) [inline cypresse2e/cypress/integration/e2eTests/aboutUs/aboutUs.spec.js]: 'Fight Pandemics' should be 'FightPandemics'

File: cypresse2e/cypress/integration/e2eTests/aboutUs/aboutUs.spec.js
Comment (jessica-lau) [inline cypresse2e/cypress/integration/e2eTests/aboutUs/aboutUs.spec.js]: 'E' in 'goToHElpBoardLink' should be lowercased

File: cypresse2e/cypress/integration/e2eTests/aboutUs/aboutUs.spec.js
Comment (jessica-lau) [inline cypresse2e/cypress/integration/e2eTests/aboutUs/aboutUs.spec.js:107]: Maybe consider moving this checkAnyKindOfLinks function to the commands.js file, so it can be reused in other tests that are checking button links?

**Review by jessica-lau — CHANGES_REQUESTED** [review]
@TimeaKovacs86 woohoo good job on your first test : )

Here are my suggestions/comments:
- We should also check the functionality of the (x) that closes the "Get Involved" modal
- Per @adorapham comments, I agree that we should check the email links on the page (two of them). But I don't know if it's necessary to check all of the supporters' links as there's a lot and probably more will be added in the future. Maybe we can do a spot check - @galeza what do you think? Other than that, I think using a function to call the class elements in the test is ok.

**Comment by TimeaKovacs86** [comment]
Thank you for your suggestions @jessica-lau , @adorapham . The check of the email and the  exit button on the modal window are good catches! I asked @galeza before I started this test suit if I should check the supporter links or not...and we agreed in not to test them. ( "no, it doesn't work with cypress. you won't be able to verify that. It will open in a new window and cypress will not allow you to switch") Also It would be difficult cause as I checked there are no selectors for those links which means I would have to write xpath localization, and the tests would be fragile. I will correct the code where you suggested! And write the missing ones. Thank you again!

File: cypresse2e/cypress/integration/e2eTests/aboutUs/aboutUs.spec.js
Comment (TimeaKovacs86) [reply cypresse2e/cypress/integration/e2eTests/aboutUs/aboutUs.spec.js:107]: Done! :D

File: cypresse2e/cypress/elements/pages/aboutUs.js
Comment (TimeaKovacs86) [reply cypresse2e/cypress/elements/pages/aboutUs.js]: Done!

File: cypresse2e/cypress/integration/e2eTests/aboutUs/aboutUs.spec.js
Comment (TimeaKovacs86) [reply cypresse2e/cypress/integration/e2eTests/aboutUs/aboutUs.spec.js]: Done!

File: cypresse2e/cypress/integration/e2eTests/aboutUs/aboutUs.spec.js
Comment (TimeaKovacs86) [reply cypresse2e/cypress/integration/e2eTests/aboutUs/aboutUs.spec.js]: Done!

File: cypresse2e/cypress/integration/e2eTests/aboutUs/aboutUs.spec.js
Comment (TimeaKovacs86) [reply cypresse2e/cypress/integration/e2eTests/aboutUs/aboutUs.spec.js:107]: Done!

**Comment by jessica-lau** [comment]
@TimeaKovacs86 thanks for making the changes! All tests passed 👍. 
I noticed the format of the code is single quoted vs double quoted. Just sent you a DM on slack.

**Comment by galeza** [comment]
Hi @TimeaKovacs86 

One test needs to be updated. We changed value in the href ContactUs:
Screen Shot 2020-11-16 at 13 37 24
In the POM we have:
`  contactEmailHref = "mailto:admin@fightpandemics.com";`
PR #1811 has been merged recently.

**Comment by galeza** [comment]
@TimeaKovacs86 @jessica-lau 

For supporters'  links - maybe we can check  lifetime, current and past supporters containers exist:
```
main > div > div > div:nth-child(4)
main > div > div > div:nth-child(7)
main > div > div> div:nth-child(10)
```
and check their visibility. There is a possibility to call children() method and check number of children (<a/>) but this number will change (I think it will be updated).
`cy.get('.left-nav>.nav').children().should('have.length', 8)`

File: cypresse2e/cypress/elements/pages/aboutUs.js
Comment (galeza) [inline cypresse2e/cypress/elements/pages/aboutUs.js]: CHange to contact@@fightpandemics.com

Comment (galeza) [reply cypresse2e/cypress/elements/pages/aboutUs.js]: @TimeaKovacs86 

I was thinking also about moving some links to fixtures (like: social media, emails). They can be used across tests.

Comment (jessica-lau) [reply cypresse2e/cypress/elements/pages/aboutUs.js]: Yea @galeza this is a good idea. A json file named something like socialContactInfo can be created in the fixtures folder.

Comment (TimeaKovacs86) [reply cypresse2e/cypress/elements/pages/aboutUs.js]: Done, created a socialContactInfo.json file in the fixtures directory, which contains the social media links and the partnership and contact emails. It goes up with the next commit!

File: cypresse2e/cypress/elements/pages/aboutUs.js
Comment (jessica-lau) [inline cypresse2e/cypress/elements/pages/aboutUs.js]: @TimeaKovacs86 I know @galeza had mentioned earlier that we can also check the members but the number of supporters will likely keep increasing overtime and also move between the lifetime/current/past sections. I think just checking that the supporter containers exist should be enough so that we don't have to keep updating the number when new ones are added.

**Comment by TimeaKovacs86** [comment]
Regarding to @jessica-lau's question about the supporter links test, could someone confirm it for me if we need those tests or not? Either way, I am ok with the team's decision. Let me know what you think. @galeza @adorapham @nivetheatha @idjevm 


@TimeaKovacs86 , we need to test if the three containers for lifetime/current/past sections are visible.  We don't need to check on links on these containers.  For the children (links on each container) will be changed when moving between these containers , so we might skip it.  That is what I understand basing on the above comments and I agree  (Adora Pham)

File: cypresse2e/cypress/elements/pages/aboutUs.js
Comment (galeza) [reply cypresse2e/cypress/elements/pages/aboutUs.js]: @jessica-lau @TimeaKovacs86 
ok, agree

**Comment by galeza** [comment]
@jessica-lau @TimeaKovacs86 
One thing is interesting for me. We used to use cy.get(selector) methods in the POM file like in the joinNow.js:
```
  getEmailField() {
    return cy.get(this.emailField);
  }
```
Here we have only selectors in the POM aboutUs.js and operations on page were moved to test spec file like:
```
  function closeGetInvolvedModalWindow() {
    **cy.get(aboutUs.getInvolvedCloseButton)**.click();
  }
```

I always put all operations and searching for elements in the POM file. Maybe this is selenium inheritance but maybe all operations on elements should we put in POM? What do you think @jessica-lau ?

Sorry, I didn't see this comment before I committed the last changes on the code. So, do you prefer if I move the getInvolvedCloseButton as a function in the POM aboutUs.js and in the spec there will be:
```
 function closeGetInvolvedModalWindow() {
    getInvolvedCloseButton().click()
}
```
And if I understand good this is the only thing what I should move to aboutUs.js, the last two functions written in the spec (checkEmailWorks ()and supporterContainerVisible()) are good where they are? Or should I move the one with the cy.xpath to the POM as well?

**Comment by TimeaKovacs86** [comment]
Hi everyone! I took out those test that checked the members of each supporter group. As far as I know the tests are running and passing. Let me know if I can do something else to make it better... Thank you for all the help!

**Comment by adorapham** [comment]
> Hi everyone! I took out those test that checked the members of each supporter group. As far as I know the tests are running and passing. Let me know if I can do something else to make it better... Thank you for all the help!

@TimeaKovacs86 , i make a quick run and get error , please take a look
image

**Comment by adorapham** [comment]
@TimeaKovacs86 , I run local and see all the test passed  but let wait for confirmation if you need to move the functions to POM file :)

**Comment by jessica-lau** [comment]
> @jessica-lau @TimeaKovacs86
> One thing is interesting for me. We used to use cy.get(selector) methods in the POM file like in the joinNow.js:
> 
> ```
>   getEmailField() {
>     return cy.get(this.emailField);
>   }
> ```
> 
> Here we have only selectors in the POM aboutUs.js and operations on page were moved to test spec file like:
> 
> ```
>   function closeGetInvolvedModalWindow() {
>     **cy.get(aboutUs.getInvolvedCloseButton)**.click();
>   }
> ```
> 
> I always put all operations and searching for elements in the POM file. Maybe this is selenium inheritance but maybe all operations on elements should we put in POM? What do you think @jessica-lau ?
> 
> Sorry, I didn't see this comment before I committed the last changes on the code. So, do you prefer if I move the getInvolvedCloseButton as a function in the POM aboutUs.js and in the spec there will be:
> 
> ```
>  function closeGetInvolvedModalWindow() {
>     getInvolvedCloseButton().click()
> }
> ```
> 
> And if I understand good this is the only thing what I should move to aboutUs.js, the last two functions written in the spec (checkEmailWorks ()and supporterContainerVisible()) are good where they are? Or should I move the one with the cy.xpath to the POM as well?

@TimeaKovacs86 yea I do agree with @galeza . Getter methods should be in the POM file so that it's more object oriented and it'll be consistent with how all the other POM files are written. You can take a look at the other POM files as an example - methods are created on the selectors and then the methods are used in the test specs instead of directly referencing to the selectors.

I came across one failed test. You'll just need to update with the latest changes from staging.

image

@jessica-lau @TimeaKovacs86 if we pull the latest code from staging and merge to this branch , more files will get updated and i think it might be similar with the issue on PR#1768, just my concern.  I see the same issue as @jessica-lau but i pull the latest code from the staging on my local to make it working.  I am not sure if we have a better solution for this (Adora Pham)

Yes I saw that test fail, and I realised it was because of the change in the email href, I also was worried about the same merging issue you has with the Privacy pull request, so just for the time I was running the tests I changed the "baseUrl" in the cypress.json file to the staging url and run the tests like that... That means once this pull request is merged then that test will pass as well. 

@jessica-lau =Good Idea 👍

**Comment by TimeaKovacs86** [comment]
I moved the cy.get and cy.xpath methods to the POM file. Please have a look, and let me know if I there is anything else I can do to make this test better.

**Review by galeza — APPROVED** [review]

**Review by jessica-lau — APPROVED** [review]
Thanks for making changes @TimeaKovacs86

**Comment by TimeaKovacs86** [comment]
Thank you all, for the help! @idjevm Please merge!

**Review by adorapham — APPROVED** [review]
LGTM. TY

**Comment by jessica-lau** [comment]
@idjevm please merge : ) Thanks.

**Review by idjevm — APPROVED** [review]

**Comment by galeza** [comment]
I see that Merge option is enabled for me so I'm merging this PR

**Pull request merged**
